### ISSUE TITLE
Looks like filterType doesn't work

### ISSUE THREAD

**Issue by homm** [hhatto/pgmagick#33]
These two images are produced using different filters, but they are absolutely the same.

![_out 320 catrom](https://user-images.githubusercontent.com/128982/31321153-dd43840e-ac89-11e7-9983-59d7d21588fd.png) ![_out 320 triangle](https://user-images.githubusercontent.com/128982/31321154-dd696f0c-ac89-11e7-9408-0ba3a1319610.png)

Looks like `.filterType` doesn't work at all.

```python
from pgmagick import Image, FilterTypes

im = Image('pineapple.jpeg')
im.filterType(FilterTypes.CatromFilter)
im.scale('320x320')
im.write('_out.320.catrom.png')

im = Image('pineapple.jpeg')
im.filterType(FilterTypes.TriangleFilter)
im.scale('320x320')
im.write('_out.320.triangle.png')
```

**Comment by komackaj** [comment]
Thank you for the report, I was able to reproduce using attached image and scaling it to 180x100

GM convert 
```bash
gm convert -filter catrom pineapple.jpeg -size "180x100" gm_catrom.png
gm convert -filter triangle pineapple.jpeg -size "180x100" gm_triangle.png
```
and GraphicsMagick++ used this way
```cpp
#include "Magick++.h"

using namespace Magick;

int main() {
    InitializeMagick(NULL);
    Image img;
    img.read("pineapple.jpeg");
    img.filterType(CatromFilter);
    img.scale("180x100");
    img.write("cpp.png");
}
```
compiled and run using
```bash
g++ $(pkg-config --cflags GraphicsMagick++) $(pkg-config --libs GraphicsMagick++) test.cpp -o test && test
```
gave different results - C++ version is same as pgmagick
```bash
md5sum *.png                                                                                              
50743fc2a1741bcc04dcdaf151598e20  cpp.png
e1bf2f7e76fbf8fcab5436c6221eaba8  gm_catrom.png
9f9d026a5acf49dcf113f98624cbb11a  gm_triangle.png
50743fc2a1741bcc04dcdaf151598e20  pgmagick.png
```

Tested on 1.3.20. Probably bad usage of GraphicsMagick++ wrapper or there is a bug in GM's cpp wrapper.

**Comment by homm** [comment]
Does it also means that `CatromFilter` is always used in pgmagick?

**Comment by komackaj** [comment]
@homm 
No, IMHO pgmagick only fails to change resampling filter, so the default one is always used. According to [GM docs](http://www.graphicsmagick.org/Magick++/Image.html#filtertype) the default filter is Lanczos.

**Comment by homm** [comment]
For some reasons can't compile the example on Ubuntu 16.04 and 14.04.

```
$ pkg-config --cflags GraphicsMagick++
-I/usr/include/GraphicsMagick

$ pkg-config --libs GraphicsMagick++
-lGraphicsMagick++ -lGraphicsMagick

$ g++ $(pkg-config --cflags GraphicsMagick++) $(pkg-config --libs GraphicsMagick++) test.cpp -o test && test
/tmp/ccsjMLfq.o: In function `main':
test.cpp:(.text+0x21): undefined reference to `Magick::InitializeMagick(char const*)'
test.cpp:(.text+0x30): undefined reference to `Magick::Image::Image()'
test.cpp:(.text+0x73): undefined reference to `Magick::Image::read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
test.cpp:(.text+0xa5): undefined reference to `Magick::Image::filterType(MagickLib::FilterTypes)'
test.cpp:(.text+0xb9): undefined reference to `Magick::Geometry::Geometry(char const*)'
test.cpp:(.text+0xd2): undefined reference to `Magick::Image::scale(Magick::Geometry const&)'
test.cpp:(.text+0xe1): undefined reference to `Magick::Geometry::~Geometry()'
test.cpp:(.text+0x11e): undefined reference to `Magick::Image::write(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
test.cpp:(.text+0x176): undefined reference to `Magick::Image::read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
test.cpp:(.text+0x1a5): undefined reference to `Magick::Image::filterType(MagickLib::FilterTypes)'
test.cpp:(.text+0x1b9): undefined reference to `Magick::Geometry::Geometry(char const*)'
test.cpp:(.text+0x1d2): undefined reference to `Magick::Image::scale(Magick::Geometry const&)'
test.cpp:(.text+0x1e1): undefined reference to `Magick::Geometry::~Geometry()'
test.cpp:(.text+0x21e): undefined reference to `Magick::Image::write(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
test.cpp:(.text+0x248): undefined reference to `Magick::Image::~Image()'
test.cpp:(.text+0x2a2): undefined reference to `Magick::Geometry::~Geometry()'
test.cpp:(.text+0x303): undefined reference to `Magick::Geometry::~Geometry()'
test.cpp:(.text+0x33c): undefined reference to `Magick::Image::~Image()'
collect2: error: ld returned 1 exit status
```

Could you help?

**Comment by homm** [comment]
> According to GM docs the default filter is Lanczos.

It looks like this is not the case. While default filter for `gm convert` is Lanczos:

```
$ compare -metric RMSE gm_lanczos.png gm_default.png NULL:
0 (0)
```

Default filter for pgmagick is different:

```
$ compare -metric RMSE gm_lanczos.png pgmagick_default.png NULL:
652.363 (0.00995442)
```

And this difference is more than between Lanczos and Catrom:

```
$ compare -metric RMSE gm_lanczos.png gm_catrom.png NULL:
357.292 (0.00545193)
```

But less athan between Lanczos and Triangle:

```
$ compare -metric RMSE gm_lanczos.png gm_triangle.png NULL:
901.662 (0.0137585)
```

**Comment by komackaj** [comment]
@homm 
Sorry, correct command line should have been (link flags after -o)
```bash
g++ $(pkg-config --cflags GraphicsMagick++) test.cpp -o test $(pkg-config --libs GraphicsMagick++) && ./test
```

Many thanks for such a deep problem analysis, I will look into this as soon as I will have some spare time

**Comment by homm** [comment]
Looks like `.scale()` isn't convolution resize method at all and doesn't use filters. From [the docs](http://www.graphicsmagick.org/Magick++/Image.html#scale):

> scale
Resize image by using simple ratio algorithm which provides good quality:

Only `.resize()` and `.zoom()` in earlier versions are convolution-based. Most likely `.scale()` method uses [supersampling resizing](https://stackoverflow.com/questions/12526449/what-is-the-name-of-this-algorithm-and-how-does-it-compare-to-other-image-resam). In support of this theory images upscaled with `.scale()` methods are pixelated like with simple nearest neighbor method. 

I think `.scale()` should be replaced with `.zoom()` (because it presents in every version) in some places:

[pgmagick.api.Image.scale](https://github.com/hhatto/pgmagick/blob/master/pgmagick/api.py#L720)
[example/pgmagick_prof.py](https://github.com/hhatto/pgmagick/blob/49ae1e5d56f52f4dbc93b8b3ba8aedc9618ed7dc/example/pgmagick_prof.py)

**Comment by komackaj** [comment]
@homm Nicely done. Using `.zoom()` in your code sample produces same results as gm, fixed command for gm is (`-resize` instead of `-size`)  
`gm convert pineapple.jpeg -filter catrom -resize 180x100 catrom_gm.png`

Upvoting for using `.zoom` (or `.resize` if present?) over `.scale` in `api.Image` - at least when a filter is specified (otherwise .scale's quality/performace ratio is IMHO an optimal solution).
@hhatto What's your opinion on this?

**Comment by hhatto** [comment]
I think that `resize` method should be used in `pgmagick.api.Image.scale` too.
I will change the code.

Thanks for suggestion.

FYI:
`Image.zoom()` is calling `Image.resize()` internally.
http://hg.code.sf.net/p/graphicsmagick/code/file/c8a67d5ba503/Magick%2B%2B/lib/Image.cpp#l2170

**Comment by komackaj** [comment]
Should be fixed by #34, closing. Feel free to reopen if needed

**Comment by homm** [comment]
Thanks!
